marketplace-builder/src/builder.rs

```diff
@@ -332,54 +360,79 @@ mod test {
         let builder_port = pick_unused_port().expect("No ports free");
         let builder_api_url: Url = format!("http://localhost:{builder_port}").parse().unwrap();
 
-        // Set up and start the network
-        let anvil = Anvil::new().spawn();
-        let l1 = anvil.endpoint().parse().unwrap();
-        let network_config = TestConfigBuilder::default().l1_url(l1).build();
+        (
+            (query_port, event_port),
+            (query_api_url, event_service_url, builder_api_url),
+        )
+    }
 
-        let tmpdir = TempDir::new().unwrap();
+    /// Initiate a mock solver and register a rollup.
+    ///
+    /// Returns the solver and its base URL.
+    async fn init_mock_solver_and_register_rollup() -> (MockSolver, Url) {
+        let mock_solver = MockSolver::init().await;
+        let solver_base_url = mock_solver.solver_url.clone();
+        let client = connect_to_solver(solver_base_url.clone());
+
+        // Create a list of signature keys for rollup registration data
+        let mut signature_keys = Vec::new();
+
+        for _ in 0..10 {
+            let private_key =
+                <BLSPubKey as SignatureKey>::PrivateKey::generate(&mut rand::thread_rng());
+            signature_keys.push(BLSPubKey::from_private(&private_key))
+        }
 
-        let config = TestNetworkConfigBuilder::default()
-            .api_config(
-                Options::with_port(query_port)
-                    .submit(Default::default())
-                    .query_fs(
-                        Default::default(),
-                        persistence::fs::Options::new(tmpdir.path().to_owned()),
-                    )
-                    .hotshot_events(HotshotEvents {
-                        events_service_port: event_port,
-                    }),
-            )
-            .network_config(network_config)
-            .build();
-        let _network = TestNetwork::new(config, MockSequencerVersions::new()).await;
+        let private_key =
+            <BLSPubKey as SignatureKey>::PrivateKey::generate(&mut rand::thread_rng());
+        let signature_key = BLSPubKey::from_private(&private_key);
```

Comment:
It might just be how GitHub is displaying the diff, but these seems like the same logic that was at line 381? What is the purpose of the logic here?

Reply:
`signature_keys` is the list of keys that can update the registration, while the `signature_key` without "s" is the key to sign this registration. Alternatively, I could just create a list of keys and use one of them as the `signature_key`, but I was using the same implementation as in the mock solver test.

Now that I think about this, this helper function could be helpful for both builder and solver tests, so I'm going to simplify the logic here and move the function to the solver crate.

Reply:
> Alternatively, I could just create a list of keys and use one of them as the `signature_key`, but I was using the same implementation as in the mock solver test.

I realized that this alternative approach wasn't really better than the current implementation, since we would need to keep track of the index in the list and make a copy of the private key associated with `signature_key`. The current implementation, though looking a bit redundant, actually does the minimal work needed. I'll just keep it as is unless you think it could be optimized. 😃

Reply:
Makes sense!